src/tpm/android/knox/java/CryptoManager.java

```diff
@@ -0,0 +1,423 @@
+ *
+ */
+public class CryptoManager {
+    private static final String ANDROID_KEY_STORE = "AndroidKeyStore";
+    private final KeyStore keyStore;
+    private String KEY_NAME;
+
+    /**
+     * Constructs a new instance of {@code CryptoManager} with the default Android KeyStore.
+     * <p>
+     * This constructor initializes the {@code CryptoManager} with the default Android KeyStore. The Android KeyStore
+     * provides a secure storage facility for cryptographic keys and certificates. Upon construction, the key store is
+     * initialized, enabling the {@code CryptoManager} to interact with cryptographic keys securely stored on the
+     * Android device. If the initialization of the key store fails, a {@link KeyStoreException} is thrown, indicating
+     * issues with the key store setup process.
+     *
+     * @throws KeyStoreException if the KeyStore Provider does not exist or fails to initialize, indicating issues with
+     *                           the key store setup process.
+     */
+    public CryptoManager() throws KeyStoreException {
+        keyStore = KeyStore.getInstance(ANDROID_KEY_STORE);
+    }
+
+    /**
+     * Generates a new symmetric key and saves it into the Android KeyStore.
+     * <p>
+     * This method initializes a new symmetric key for encryption and decryption purposes using the specified symmetric key algorithm.
+     * The key is stored in the Android KeyStore.
+     * Additionally, this method ensures that the key is backed by the strong box feature.
+     *
+     * @param key_id     The unique identifier under which the key will be stored in the KeyStore.
+     * @param keyGenInfo A string containing key generation parameters separated by semicolons. Expected format: "KEY_ALGORITHM;KEY_SIZE;BLOCK_MODE;PADDING".
+     * @throws CertificateException               if there is an issue loading the certificate chain.
+     * @throws IOException                        for I/O errors such as incorrect passwords.
+     * @throws NoSuchAlgorithmException           if the generation algorithm does not exist or the keystore doesn't exist.
+     * @throws NoSuchProviderException            if the provider does not exist.
+     * @throws InvalidAlgorithmParameterException for invalid or nonexistent parameters.
+     * @throws KeyStoreException                  if there is an error accessing the keystore.
+     */
+    public void genKey(String key_id, String keyGenInfo) throws CertificateException,
+            IOException, NoSuchAlgorithmException, NoSuchProviderException,
+            InvalidAlgorithmParameterException, KeyStoreException {
+        String[] keyGenInfoArr = keyGenInfo.split(";");
+        String KEY_ALGORITHM = keyGenInfoArr[0];
+        int KEY_SIZE = Integer.parseInt(keyGenInfoArr[1]);
+        String BLOCKING = keyGenInfoArr[2];
+        String PADDING = keyGenInfoArr[3];
+
+        KEY_NAME = key_id;
+        keyStore.load(null);
```

Comment:
Why is this needed?

Reply:
We need to load the keystore before storing the key inside of it, otherwise the keystore will throw an exception.
We set KEY_NAME = key_id, as we thought it will enhance readablillity, future-proofing and conformity.
Does this answer your question?